handlers/changes.js

```diff
@@ -170,6 +170,11 @@ var getChanges = function(feed) {
     body: { doc_ids: _.union(feed.requestedIds, feed.validatedIds) },
     method: 'POST'
   }, function(err, changes) {
+    if (feed.res.finished) {
+      // don't write to the response if it has already ended
+      return;
+    }
+    cleanUp(feed);
```

Comment:
Is this moved up here to try avoiding reusing the same feed? 
The `abort` l. 159 looks like it would cause trouble in case of `!err`, since you haven't written to the feed yet? (don't actually know what `abort` does, maybe not)